src/main/java/org/folio/entlinks/controller/delegate/AuthoritySourceFileServiceDelegate.java

```diff
@@ -167,4 +177,17 @@ public boolean anyAuthoritiesExistForSourceFile(AuthoritySourceFile sourceFile)
 
     return false;
   }
+
+  @NotNull
+  private BiConsumer<AuthoritySourceFile, AuthoritySourceFile> updateConsumer() {
+    return (newAsf, oldAsf) -> {
+      eventPublisher.publishUpdateAsfEvent(mapper.toDto(oldAsf), mapper.toDto(newAsf));
+      propagationService.propagate(newAsf, UPDATE, context.getTenantId());
```

Comment:
It's already propagated in the end of `patchAuthoritySourceFile`. Task said "also produce on propagation" which means you need not only send kafka event on main logic, but also inside propagation logic